lib/vector/diglib/plus_struct.c

```diff
@@ -673,13 +673,68 @@ int dig_Wr_Plus_head(struct gvfile *fp, struct Plus_head *ptr)
     if (0 >= dig__fwrite_port_C((char *)buf, 5, fp))
         return (-1);
 
-    /* determine required offset size from coor file size */
-    if (ptr->coor_size > (off_t)PORT_LONG_MAX) {
-        /* can only happen when sizeof(off_t) == 8 */
-        ptr->off_t_size = 8;
```

Comment:
Please don't remove this test. `ptr->off_t_size` must still be set to 8 if the coor file is larger than `PORT_LONG_MAX`, no matter how large the topo file would become.

Suggestion: First check the coor file size, if `ptr->off_t_size` is not set to 8, use your new check if the topo file would become larger than `PORT_LONG_MAX`.

Reply:
Yep! I missed that. Obviously, `ptr->off_t_size` is shared between coor and topo files as of now.
https://github.com/OSGeo/grass/blob/6ffd18f8083c888fddfdae86ceb771e0e33cbb25/lib/vector/diglib/plus_struct.c#L762

I cannot see anywhere else `ptr->off_t_size` is used for coor files other than the above line and the reading part where its size is written and read. If we use a different-size variable for the coor size in these two lines, theoretically, we can only save 4 bytes only if coor file is less than `PORT_LONG_MAX`, but topo file is greater than that.